x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfig.java

```diff
@@ -503,6 +529,9 @@ public Builder setMultivariateByFields(Boolean multivariateByFields) {
          */
         public AnalysisConfig build() {
             TimeUtils.checkPositiveMultiple(bucketSpan, TimeUnit.SECONDS, BUCKET_SPAN);
+            if (modelPruneWindow != null) {
+                TimeUtils.checkNonNegativeMultiple(modelPruneWindow, TimeUnit.SECONDS, MODEL_PRUNE_WINDOW);
+            }
```

Comment:
Shouldn't this always be greater than the BucketSpan?

Reply:
Two times the bucket span is the smallest value that makes sense, so that should be enforced here.

Since this gets converted into a number of buckets in the C++ code, we should possibly also enforce that the value specified is an exact number of buckets.  That then means the functions in `TimeUtils` aren't much use for enforcing the value.  It would need a bespoke check and error message.